lib/types.nix

```diff
@@ -113,4 +113,32 @@ rec {
         types.optionDescriptionPhrase (class: class == "noun" || class == "composite") elemType
       }";
     };
+
+  pluginLuaConfig = types.submodule (
+    { config, ... }:
+    {
+      options = {
+        pre = lib.mkOption {
+          type = types.lines;
+          default = "";
+          description = "Configuration before the initialization of the plugin";
+        };
+        post = lib.mkOption {
+          type = types.lines;
+          default = "";
+          description = "Configuration after the initialization of the plugin";
+        };
+        init = lib.mkOption {
+          type = types.lines;
+          default = "";
+          description = "Initialization code (between pre & post)";
+        };
+      };
+
+      config.init = lib.mkMerge [
+        (lib.mkBefore config.pre)
+        (lib.mkAfter config.post)
```

Comment:
I wonder if it better to use non-standard merge priority numbers?

E.g. what would we expect to happen when someone uses:

```
pre = "foo";
init = mkBefore "bar";
```

Currently they would both be merged at priority 500, however we can't know just from reading the definition of we'll get `"foo\nbar"` or `"bar\nfoo"`.

I think users would expect `pre` to be merged earlier than mkBefore and `post` to be merged after mkAfter?

Reply:
Definitely a niche case, but makes sense to at least prevent default priority overrides from messing with ordering.